src/runtime/internal/scheduler.ts

```diff
@@ -73,8 +73,9 @@ function update($$) {
 	if ($$.fragment !== null) {
 		$$.update();
 		run_all($$.before_update);
-		$$.fragment && $$.fragment.p($$.ctx, $$.dirty);
+		const dirty = $$.dirty;
 		$$.dirty = [-1];
```

Comment:
This will cause update to refresh everything, whether changed or not.
Also, we need to be sure that the action callback is only triggered once.

Reply:
I thought of another oops... what if the 32nd element is invalidated?
We'd lose the bit since the special `[-1]` is not handled inside `p`.

Reply:
only have 31 bit per item in the array, if i understand correctly,
https://github.com/sveltejs/svelte/blob/master/src/compiler/compile/render_dom/Renderer.ts#L216-L217

Reply:
@tanhauhau,

I think I'm losing it because I thought I saw the fragment update called twice but it is only once.... so nevermind, you are absolutely correct here.

Since this change exists in the other PR, we should close this out.